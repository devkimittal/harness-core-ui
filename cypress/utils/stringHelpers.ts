export function getIdentifierFromName(str: string): string {
  return str
    .trim()
    .replace(/[^0-9a-zA-Z_$ ]/g, '') // remove special chars except _ and $
    .replace(/ +/g, '_') // replace spaces with _
}
