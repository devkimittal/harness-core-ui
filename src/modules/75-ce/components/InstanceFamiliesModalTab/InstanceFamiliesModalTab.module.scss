/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

table {
  border-spacing: 0;

  .cell:nth-child(odd) {
    background-color: var(--grey-100);
  }
}

.headerGroup {
  th {
    width: auto;
    padding: var(--spacing-small);
  }
}

.tableHeader {
  width: max-content;
}
