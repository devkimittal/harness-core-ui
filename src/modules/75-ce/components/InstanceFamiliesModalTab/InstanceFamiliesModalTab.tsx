/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useMemo } from 'react'
import { Container, Text, Color, FontVariation, Checkbox, TableV2 } from '@harness/uicore'
import type { Column } from 'react-table'

import { flatten } from 'lodash-es'
import { Action, ACTIONS, IState } from '../NodeRecommendation/NodeRecommendation'

enum CheckboxStatus {
  CHECKED = 'checked',
  UNCHECKED = 'unchecked',
  INDETERMINATE = 'indeterminate'
}

interface TabProps {
  data?: Record<string, string[]>
  state: IState
  dispatch: React.Dispatch<Action>
}

export const InstanceFamiliesModalTab: React.FC<TabProps> = ({ data, state, dispatch }) => {
  const categoryData = data || {}

  const series = Object.keys(categoryData).sort()
  const types = flatten(Object.values(categoryData))

  const getCheckboxStatus = (val: string): CheckboxStatus => {
    if (state.includeTypes.includes(val)) {
      return CheckboxStatus.CHECKED
    } else if (state.excludeTypes.includes(val)) {
      return CheckboxStatus.INDETERMINATE
    } else {
      return CheckboxStatus.UNCHECKED
    }
  }

  const formatData = (seriesList: string[], typesList: string[]): Record<string, boolean>[] => {
    const formattedData: Record<string, boolean>[] = []

    seriesList.map(() => {
      const d: Record<string, boolean> = {}

      typesList.map(t => (d[t] = state.includeTypes.includes(t)))

      formattedData.push(d)
    })

    return formattedData
  }

  const columns: Column<Record<string, boolean>>[] = useMemo(
    () => [
      {
        Header: '',
        id: 'series',
        width: '50px',
        Cell: ({ row }: { row: any }) => <Text font={{ variation: FontVariation.SMALL_SEMI }}>{series[row.index]}</Text>
      },
      ...types.map(type => ({
        Header: type,
        accessor: type,
        id: type,
        Cell: ({ row }: { row: any }) => {
          const status = getCheckboxStatus(type)

          if (!categoryData[series[row.index]].includes(type)) {
            return null
          }

          return (
            <Container flex={{ justifyContent: 'center', alignItems: 'center' }} background={row.index}>
              <Checkbox
                checked={status === CheckboxStatus.CHECKED}
                indeterminate={status === CheckboxStatus.INDETERMINATE}
                onClick={() => {
                  console.log(status)
                  if (status === CheckboxStatus.UNCHECKED) {
                    dispatch({ type: ACTIONS.INCLUDE_TYPES, data: type })
                  } else if (status === CheckboxStatus.CHECKED) {
                    dispatch({ type: ACTIONS.INCLUDE_TYPES, data: type })
                    dispatch({ type: ACTIONS.EXCLUDE_TYPES, data: type })
                  } else {
                    dispatch({ type: ACTIONS.EXCLUDE_TYPES, data: type })
                  }
                }}
              />
            </Container>
          )
        }
      }))
    ],
    [state]
  )

  return (
    <Container background={Color.GREY_50} flex style={{ overflow: 'scroll', border: '1px solid #D9DAE6' }}>
      <TableV2 minimal data={formatData(series, types)} columns={columns} />
    </Container>
  )
}
