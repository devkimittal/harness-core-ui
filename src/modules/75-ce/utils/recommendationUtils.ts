/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export const addBufferToValue = (value: number, bufferPercentage: number): number =>
  +(((100 + bufferPercentage) / 100) * value).toFixed(2)

export const calculateSavingsPercentage = (savings: number, totalCost: number): string =>
  `(${Math.floor((savings / totalCost) * 100)}%)`

export const isResourceConsistent = (sumCpu: number, sumMemory: number, maxCpu: number, maxMemory: number): boolean => {
  const isInconsistent = Math.round(sumCpu) < Math.round(maxCpu) || Math.round(sumMemory) < Math.round(maxMemory)

  const anyZero =
    Math.round(sumCpu) == 0 || Math.round(sumMemory) == 0 || Math.round(maxCpu) == 0 || Math.round(maxMemory) == 0

  return !isInconsistent || !anyZero
}

export const calculateNodes = (
  sumCpu: number,
  sumMemory: number,
  maxCpu: number,
  maxMemory: number,
  minNodes: number
): { maximumNodes: number; minimumNodes: number } => {
  let minimumNodes = minNodes || 3

  let maximumNodes = Math.min(Math.floor(sumCpu / maxCpu), Math.floor(sumMemory / maxMemory))
  maximumNodes = Math.max(maximumNodes, 1)

  if (maximumNodes < minNodes) {
    minimumNodes = maximumNodes
  }

  return { maximumNodes, minimumNodes }
}
