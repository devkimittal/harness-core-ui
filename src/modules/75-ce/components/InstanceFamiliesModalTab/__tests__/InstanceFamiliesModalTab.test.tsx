/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'

import MockInstanceFamilyData from '@ce/components/NodeRecommendation/__tests__/MockInstanceFamilyData.json'
import { InstanceFamiliesModalTab } from '../InstanceFamiliesModalTab'

const mockState = {
  maxCpu: 44,
  maxMemory: 100,
  sumCpu: 30,
  sumMem: 120,
  minNodes: 10,
  includeTypes: [],
  includeSeries: [],
  excludeTypes: [],
  excludeSeries: []
}

describe('test cases for instance family modal tab', () => {
  test('should be able to render instance family modal tab', async () => {
    const { container } = render(
      <TestWrapper>
        <InstanceFamiliesModalTab
          dispatch={jest.fn()}
          state={mockState}
          data={MockInstanceFamilyData.categoryDetails['Compute optimized']}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
