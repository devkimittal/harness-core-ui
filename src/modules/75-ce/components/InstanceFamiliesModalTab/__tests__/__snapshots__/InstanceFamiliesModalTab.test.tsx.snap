// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for instance family modal tab should be able to render instance family modal tab 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--flex"
    style="overflow: scroll; border: 1px solid #d9dae6;"
  >
    <table
      role="table"
    >
      <thead>
        <tr
          class="headerGroup"
          role="row"
        >
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                c2d-highcpu-4
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                c2d-highcpu-32
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                c2d-highcpu-8
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                c2d-highcpu-112
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                c2d-highcpu-2
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                c2d-highcpu-16
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                c2d-highcpu-56
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                e2-highcpu-4
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                e2-highcpu-2
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                e2-highcpu-8
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                e2-highcpu-32
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                e2-highcpu-16
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                n1-highcpu-8
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                n1-highcpu-2
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                n1-highcpu-96
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                n1-highcpu-4
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                n1-highcpu-32
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                n1-highcpu-16
              </p>
            </div>
          </th>
          <th
            class=""
            colspan="1"
            role="columnheader"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main columnCheckbox"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
              </div>
              <p
                class="StyledProps--font StyledProps--main tableHeader StyledProps--font-variation-small-bold"
              >
                n1-highcpu-64
              </p>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            class=""
            role="cell"
          >
            <div
              class="Layout--horizontal StyledProps--main rowHeader StyledProps--padding-small"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main rowCheckboxContainer"
              >
                <div
                  class="StyledProps--font StyledProps--main rowCheckbox"
                >
                  <label
                    class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                  >
                    <input
                      type="checkbox"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                  </label>
                </div>
              </div>
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
              >
                c2d
              </p>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
        </tr>
        <tr
          role="row"
        >
          <td
            class=""
            role="cell"
          >
            <div
              class="Layout--horizontal StyledProps--main rowHeader StyledProps--padding-small"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main rowCheckboxContainer"
              >
                <div
                  class="StyledProps--font StyledProps--main rowCheckbox"
                >
                  <label
                    class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                  >
                    <input
                      type="checkbox"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                  </label>
                </div>
              </div>
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
              >
                e2
              </p>
            </div>
          </td>
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
        </tr>
        <tr
          role="row"
        >
          <td
            class=""
            role="cell"
          >
            <div
              class="Layout--horizontal StyledProps--main rowHeader StyledProps--padding-small"
              style="align-items: center;"
            >
              <div
                class="StyledProps--font StyledProps--main rowCheckboxContainer"
              >
                <div
                  class="StyledProps--font StyledProps--main rowCheckbox"
                >
                  <label
                    class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                  >
                    <input
                      type="checkbox"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                  </label>
                </div>
              </div>
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
              >
                n1
              </p>
            </div>
          </td>
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          />
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
          <td
            class=""
            role="cell"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
